contracts/hackatom/tests/integration.rs

```diff
@@ -158,18 +158,18 @@ fn failed_handle() {
         verifier: verifier.clone(),
         beneficiary: beneficiary.clone(),
     };
-    let init_params = mock_params(
+    let init_env = mock_env(
         &deps.api,
         creator.as_str(),
         &coin("1000", "earth"),
         &coin("1000", "earth"),
     );
-    let init_res = init(&mut deps, init_params, init_msg).unwrap();
+    let init_res = init(&mut deps, init_env, init_msg).unwrap();
     assert_eq!(0, init_res.messages.len());
 
     // beneficiary can release it
-    let handle_params = mock_params(&deps.api, beneficiary.as_str(), &[], &coin("1000", "earth"));
-    let handle_res = handle(&mut deps, handle_params, HandleMsg {});
+    let handle_env = mock_env(&deps.api, beneficiary.as_str(), &[], &coin("1000", "earth"));
```

Comment:
Having a new environment instead of a chaning environment needs getting used to

Reply:
We move it into the function when we call it, so we cannot reuse it.
If you like to modify env, we could pass in `env.clone()` and then modify it before sending a copy to the next function.